tests/std/tests/GH_005204_regex_collating_ranges/test.cpp

```diff
@@ -0,0 +1,333 @@
+            L"[s-\u00DF]", // U+00DF LATIN SMALL LETTER SHARP S
+            "de-DE", regex_constants::collate);
+        collate_s_to_sharp_s_regex.should_search_fail(L"r");
+        collate_s_to_sharp_s_regex.should_search_match(L"s", L"s");
+        collate_s_to_sharp_s_regex.should_search_match(L"S", L"S");
+        collate_s_to_sharp_s_regex.should_search_match(L"\u00DF", L"\u00DF"); // U+00DF LATIN SMALL LETTER SHARP S
+        collate_s_to_sharp_s_regex.should_search_fail(L"\u1E9E"); // U+1E9E LATIN CAPITAL LETTER SHARP S
+        collate_s_to_sharp_s_regex.should_search_fail(L"t");
+        collate_s_to_sharp_s_regex.should_search_fail(L"u");
+        for (const wchar_t* s : special_characters_without_sharp_s) {
+            collate_s_to_sharp_s_regex.should_search_fail(s);
+        }
+    }
+
+    {
+        test_wregex_locale collate_s_to_Sharp_S_regex(&g_regexTester, L"[s-\u1E9E]", // LATIN CAPITAL LETTER SHARP S
+            "de-DE", regex_constants::collate);
+        collate_s_to_Sharp_S_regex.should_search_fail(L"r");
+        collate_s_to_Sharp_S_regex.should_search_match(L"s", L"s");
+        collate_s_to_Sharp_S_regex.should_search_match(L"S", L"S");
+        collate_s_to_Sharp_S_regex.should_search_match(L"\u00DF", L"\u00DF"); // U+00DF LATIN SMALL LETTER SHARP S
+        collate_s_to_Sharp_S_regex.should_search_match(L"\u1E9E", L"\u1E9E"); // U+1E9E LATIN CAPITAL LETTER SHARP S
+        collate_s_to_Sharp_S_regex.should_search_fail(L"t");
+        collate_s_to_Sharp_S_regex.should_search_fail(L"u");
+        for (const wchar_t* s : special_characters_without_sharp_s) {
+            collate_s_to_Sharp_S_regex.should_search_fail(s);
+        }
+    }
+
+    {
+        test_wregex_locale collate_s_to_t_regex(&g_regexTester, L"[s-t]", "de-DE", regex_constants::collate);
+        collate_s_to_t_regex.should_search_fail(L"r");
+        collate_s_to_t_regex.should_search_match(L"s", L"s");
+        collate_s_to_t_regex.should_search_match(L"S", L"S");
+        collate_s_to_t_regex.should_search_match(L"\u00DF", L"\u00DF"); // U+00DF LATIN SMALL LETTER SHARP S
+        collate_s_to_t_regex.should_search_match(L"\u1E9E", L"\u1E9E"); // U+1E9E LATIN CAPITAL LETTER SHARP S
+        collate_s_to_t_regex.should_search_match(L"t", L"t");
+        collate_s_to_t_regex.should_search_fail(L"u");
+        for (const wchar_t* s : special_characters_without_sharp_s) {
+            collate_s_to_t_regex.should_search_fail(s);
+        }
+    }
+
+    {
+        test_wregex_locale collate_A_to_z_regex(&g_regexTester, L"[A-z]", "de-DE", regex_constants::collate);
+        collate_A_to_z_regex.should_search_fail(L"a");
+        for (wchar_t ascii_upper = L'A'; ascii_upper < 'Z'; ++ascii_upper) {
+            collate_A_to_z_regex.should_search_match(wstring(1, ascii_upper), wstring(1, ascii_upper));
+        }
+        for (wchar_t ascii_lower = L'b'; ascii_lower <= 'z'; ++ascii_lower) {

```

Comment:
The comparisons should be against L character literals, although it doesn't make a real difference.